next.config.js

```diff
@@ -0,0 +1,31 @@
+module.exports = {
+  webpack5: true,
+  webpack: config => {
+    config.resolve.fallback = { fs: false, module: false }
+
+    return config
+  },
+  async redirects() {
+    return [
+      {
+        source: '/p',
+        destination: '/projects',
+        permanent: true,
+      },
+      {
+        source: '/v2/create',
+        destination: '/create',
+        permanent: true,
+      },
+    ]
+  },
+  async rewrites() {
+    return [
+      {
+        source: '/@:projectId',
+        destination: '/v2/p/:projectId',
+      },
```

Comment:
**question** does this work with a project with a handle?

Reply:
Okay i see what's happening.

This isn't the expected behavior. If there's no project with the handle `@1`, the page should show "project not found".

I think we can remove this: wdyt?

Reply:
Hmm. I see the problem. We need to handle the `/@:handle` route somewhere....

Reply:
Unfortunately nextjs doesn't support that :( . We might need to do some re-routing logic. This is to support legacy v1 links, right? So it shouldn't be a big deal to do something similar to the de-hashing of links since it won't be long lived

Reply:
Looks like this will do it:
```
@@ -25,6 +25,10 @@ module.exports = {
         source: '/@:projectId',
         destination: '/v2/p/:projectId',
       },
+      {
+        source: '/:handle',
+        destination: '/p/:handle'
+      }
     ]
   },
   pageExtensions: ['page.tsx', 'page.ts', 'page.jsx', 'page.js'],
```

Reply:
Fixed the routing with middleware